pkg/codegen/go/gen_program.go

```diff
@@ -1167,6 +1167,10 @@ func (g *generator) lowerResourceOptions(opts *pcl.ResourceOptions) (*model.Bloc
 	if opts.ReplaceWith != nil {
 		appendOption("ReplaceWith", opts.ReplaceWith, model.NewListType(resourceType))
 	}
+	if opts.ReplacementTrigger != nil {
+		destType := model.NewOutputType(model.DynamicType)
```

Comment:
This had to be an `OutputType` in order to make the output programs unify it with `Input`... I'm not really sure what's going on here?